Sources/MapboxMaps/Style/StyleManagerProtocol.swift

```diff
@@ -0,0 +1,169 @@
+    /// - [pitch](https://docs.mapbox.com/mapbox-gl-js/style-spec/#root-pitch)
+    ///
+    /// The style default camera is re-evaluated when a new style is loaded.
+    var defaultCamera: CameraOptions { get }
+
+    /// Get or set the map style's transition options.
+    ///
+    /// By default, the style parser will attempt to read the style default
+    /// transition options, if any, falling back to an immediate transition
+    /// otherwise.
+    ///
+    /// The style transition is re-evaluated when a new style is loaded.
+    ///
+    /// - Attention:
+    ///     Overridden transition options are reset once a new style has been loaded.
+    var transition: TransitionOptions { get set }
+
+    /// Adds a new style layer given its JSON properties
+    ///
+    /// Runtime style layers are valid until they are either removed or a new
+    /// style is loaded.
+    ///
+    /// - See Also: https://docs.mapbox.com/mapbox-gl-js/style-spec/#layers
+    ///
+    /// - Parameters:
+    ///   - properties: A JSON dictionary of style layer properties.
+    ///   - layerPosition: If not empty, the new layer will be positioned according
+    ///         to `LayerPosition` parameters.
+    ///
+    /// - Throws:
+    ///     An error describing why the operation was unsuccessful.
+    func addLayer(with properties: [String: Any], layerPosition: LayerPosition?) throws
+
+    /// Adds a new style custom layer.
+    ///
+    /// Runtime style layers are valid until they are either removed or a new
+    /// style is loaded.
+    ///
+    /// - See Also: https://docs.mapbox.com/mapbox-gl-js/style-spec/#layers
+    ///
+    /// - Parameters:
+    ///   - id: Style layer id.
+    ///   - layerHost: Style custom layer host.
+    ///   - layerPosition: If not empty, the new layer will be positioned according
+    ///         to `LayerPosition` parameters.
+    ///
+    /// - Throws:
+    ///     An error describing why the operation was unsuccessful.
+    func addCustomLayer(withId id: String, layerHost: CustomLayerHost, layerPosition: LayerPosition?) throws
+
+    /// Removes an existing style layer
+    ///
+    /// Runtime style layers are valid until they are either removed or a new
+    /// style is loaded.
+    ///
+    /// - Parameter id: Identifier of the style layer to remove.
+    ///
+    /// - Throws:
+    ///     An error describing why the operation was unsuccessful.
+    func removeLayer(withId id: String) throws
+
+    /// Checks whether a given style layer exists.
+    ///
+    /// Runtime style layers are valid until they are either removed or a new
+    /// style is loaded.
+    ///
+    /// - Parameter id: Style layer identifier.
+    ///
+    /// - Returns: `true` if the given style layer exists, `false` otherwise.
+    func layerExists(withId id: String) -> Bool
+
+    /// The ordered list of the current style layers' identifiers and types
+    var layerIdentifiers: [LayerInfo] { get }
```

Comment:
How about renaming this to `allLayerIdentifiers` ?

Reply:
I'd be good with that. Will tackle in next PR.